compliance_checker/ioos.py

```diff
@@ -754,10 +756,10 @@ def check_creator_and_publisher_type(self, ds):
 
         return result_list
 
-    def check_platform_global(self, ds):
+    def check_platform_id(self, ds):
         """
-        The "platform" attribute must be a single string containing
-        no blank characters.
+        The "platform_id" attribute should be present if the "platform"
+        attribute is also present.
```

Comment:
Good catch. If this is the case, it doesn't seem like there are any restrictions on what the `platform_id` must be. From the examples, it looks like the value should be an alphanumeric string:

> Examples:
>   - `platform_id = "carquinez"`
>   - `platform_id = "cb0102"`

but this isn't explicitly stated.

@mwengren is the intent for the value to be an alphanumeric string? I think I placed this check in here because I was a bit confused by the statement regarding the [name of the platform variable](https://ioos.github.io/ioos-metadata/ioos-metadata-profile-v1-2.html#platform-variable):

> For the purposes of this profile - which restricts datasets to one platform-per-dataset - this variable should only have a dimension of 1 (indicating only a single sampling platform is present). Additionally, geophysical data variables must reference the name of this variable in their platform attribute (described in the Platform section above). __The Platform Variable can be of any data type, however if the global attribute platform_id is specified, it should match the value of the Platform Variable__.

I wasn't really sure what "it" was referring to in the bold above.

Reply:
`platform` and `platform_id` are independent, correct.  

As far as the bold text, I think we need to fix the documentation slightly.   As far as I understand it, the Platform Variable (as identified by the `cf_role` attributes) is intended to differentiate by sampling platform within the dataset.  At the moment, we're restricting this to dimension=1 for simplicity (possibly subject to change, unfortunately).

I don't know whether Platform Variable is typically numeric or not in most datasets - it certainly could be - but it seems too restrictive to have the values in this variable match the global `platform_id` attribute, which is meant to be more of a human-readable identifier.  I think I would remove that bolded text, probably.

Reply:
@mwengren If there are no restrictions on the type, then as @benjwadams mentioned, if this is optional, I see no reason to actually have it as a check. If there are some restrictions we can test for we can do that, otherwise we can just let it be.

Reply:
@daltonkell assuming you're talking about `platform_id`, let's just have CC note it if it's not present in the dataset in the 'Recommended' category  - I think this is what it does currently.  This is how it treats all the others that are recommended in the profile, correct?  

See: https://gist.github.com/mwengren/d0a921b5b83661b0098437bfefc05701.  This just includes: 'platform_id not present' in results.

Reply:
BTW: I just removed the text in the profile about Platform Variable and platform_id matching.  HTH.

https://ioos.github.io/ioos-metadata/ioos-metadata-profile-v1-2.html#platform-variable

Reply:
> @daltonkell assuming you're talking about `platform_id`, let's just have CC note it if it's not present in the dataset in the 'Recommended' category - I think this is what it does currently. This is how it treats all the others that are recommended in the profile, correct?
> 
> See: https://gist.github.com/mwengren/d0a921b5b83661b0098437bfefc05701. This just includes: 'platform_id not present' in results.

Typically there are still some constraints one would test against, but I'm willing to wager this comes up somewhere else in the CC.